File: 64_sap_xep_day_con_lien_tuc.py
def solve():
    global n , M , check
    M.sort()  
    for i in range(0 , n):
        if M[i]!=check[i]:
            left = i+1
            break ; 
    for i in range (n-1 , -1 , -1):
         if M[i]!=check[i]:
            right = i+1
            break ;
    print(left , right)        
t = int(input())
while(t>0):
    n = int(input())
    M = list(map(int , input().split()))
    check = []
    for i in range( 0 , n):
        check.append(M[i])     
    solve()
    t-=1

File: 64_sap_xep_day_con_lien_tuc.cpp
#include<iostream>
#include<algorithm>
using namespace std ;
void solve() {
    int n ; cin>>n ; 
    int M[n] , check[n] ; 
    for(int i=0 ; i< n ; i++) {
        cin>>M[i] ; check[i] = M[i] ; 
    }
    sort(M , M+n ) ; 
    int left , right ; 
    for(int i=0 ; i<n ; i++) {
        if(M[i]!=check[i]){
            left = i+1 ; 
            break ; 
        }
    }
    for(int i=n-1 ; i>=0 ; i--){
        if(M[i]!=check[i]){
            right = i+1 ;
            break ; 
        }
    }
    cout<<left<<" "<<right<<endl ; 
}
main(){
    int t;  cin>> t;
    while(t--) {
        solve() ; 
    }
}